src/main/java/com/uid2/shared/attest/UidCoreClient.java

```diff
@@ -104,20 +104,14 @@ private InputStream readContentFromLocalFileSystem(String path, Proxy proxy) thr
         return (proxy == null ? new URL(path).openConnection() : new URL(path).openConnection(proxy)).getInputStream();
     }
 
-    private InputStream getWithAttest(String path, String jwtToken) throws IOException, InterruptedException, AttestationTokenRetrieverException {
-        if (!attestationTokenRetriever.attested()) {
-            attestationTokenRetriever.attest();
-        }
-
+    private InputStream get(String path, String jwtToken) throws IOException, InterruptedException {
         String attestationToken = attestationTokenRetriever.getAttestationToken();
 
         HttpResponse<String> httpResponse;
         httpResponse = sendHttpRequest(path, attestationToken, jwtToken);
 
-        // This should never happen, but keeping this part of the code just to be extra safe.
         if (httpResponse.statusCode() == 401) {
             LOGGER.info("Initial response from UID2 Core returned 401, performing attestation");
-            attestationTokenRetriever.attest();
```

Comment:
Why did you remove this?

Reply:
We want to decouple the attest() calls from the calls to a refresh endpoint.

"The renewal of the attestation process should be changed to be done as a background process, rather than being done as part of a request - response call from a client of the Operator. This would allow for the retries to not block an incoming request."

Calls to attest are now call handled within the AttestationTokenRetreiver class

Reply:
This work was already done, and attestation now happens 10 minutes before the expiry of the token. I don't believe we need to change this. getWithAttest allows the first request to perform attestation, at which point the schedule will start.

Reply:
Ah I see I was not considering that the expiration check will handle the retry already for failure in all cases except if the first attestation call fails. I have undone this change.